util/dvsim/testplanner/testplan_utils.py

```diff
@@ -132,32 +132,43 @@ def parse_hjson(filename):
 
 
 def merge_dicts(list1, list2, use_list1_for_defaults=True):
-    '''merge 2 dicts into one
+    '''Merge 2 dicts into one
 
-    This funciton takes 2 dicts as args list1 and list2. It recursively merges list2 into
+    This function takes 2 dicts as args list1 and list2. It recursively merges list2 into
     list1 and returns list1. The recursion happens when the the value of a key in both lists
     is a dict. If the values of the same key in both lists (at the same tree level) are of
     dissimilar type, then there is a conflict and an error is thrown. If they are of the same
     scalar type, then the third arg "use_list1_for_defaults" is used to pick the final one.
     '''
-    for key in list2.keys():
-        if key in list1:
-            if type(list1[key]) is list and type(list2[key]) is list:
-                list1[key].extend(list2[key])
-            elif type(list1[key]) is dict and type(list2[key]) is dict:
-                list1[key] = merge_dicts(list1[key], list2[key])
-            elif (type(list1[key]) == type(list2[key])):
-                if not use_list1_for_defaults:
-                    list1[key] = list2[key]
-            else:
-                print("The type of value of key \"", key, "\" in list1: \"", \
-                      str(type(list1[key])), \
-                      "\" does not match the type of value in list2: \"", \
-                      str(type(list2[key])), \
-                      "\". The two lists cannot be merged.")
-                sys.exit(1)
-        else:
-            list1[key] = list2[key]
+    for key, item2 in list2.items():
```

Comment:
Thanks for rewriting it, LG! Would you please move this change to a separate commit though, since it is not really a PEP8 fix?

Reply:
Well, I only looked at the code because the type comparison triggers a PEP8 error. But I'll split it out into a separate commit if you think that would be better: I'll push up a new version in a few minutes. (Hopefully that will trigger a new CI run, which might work now).

Reply:
This looks functionally equivalent to the old code, but we cannot say for sure unless we test it. Moving it to a separate commit allows an easier path to roll back if needed. Thanks for splitting it!